### PR TITLE
New module asa_ogcontent: Find object-group content

### PR DESCRIPTION
**Pull request by lvrfrc87**
##### SUMMARY
Find object-group' s contet including content of parent and child nested object-groups on Cisco ASA device.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
asa_ogcontent.py

##### ADDITIONAL INFORMATION
Playbook example:
```- name: find object-group's content.
  asa_ogcontent:
    name: aws_all_critical_vpcs
```
Return list example:
```
[
  "object-group network aws_all_critical_vpcs",
   "group-object aws_critical_west",
      "group-object aws_critical_west_ngnix",
        "network-object 10.0.160.0 255.255.248.0",
      "group-object aws_critical_west_sql",
        "network-object 10.0.168.0 255.255.248.0",
   "group-object aws_critical_est",
      "group-object aws_critical_est_apache",
        "network-object 10.0.192.0 255.255.248.0",
      "group-object aws_critical_est_postgres",
        "network-object 10.0.200.0 255.255.248.0"
    ]
```

This module run just show command and return the content of a specific object-group passed in `name` arg. All nested and children object groups are found and returned with proper indentation

### CODE REVIEW

**Review by Qalthos — CHANGES_REQUESTED** [review]
So I haven't reviewed the module extensively yet, but this PR won't get merged until there are some unit tests attached.

**Comment by FedericoOlivieri** [comment]
@Qalthos  My intention was to submit a PR and discuss with the community about this module.
If the community believes that this module can be useful and can be merged, I' ll be happy to provide all the test required.

**Comment by FedericoOlivieri** [comment]
@Qalthos did you have any chance to review this PR? :)

File: lib/ansible/modules/network/asa/asa_ogcontent.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+EXAMPLES = """
+---
+- name: find object-group's content.
+  asa_ogcontent:
+    name: aws_all_critical_vpcs
+"""
+
+RETURN = """
+commands:
```
Comment (Qalthos) [inline lib/ansible/modules/network/asa/asa_ogcontent.py:39]: I don't know that `commands` is the right key for this. In other modules, `commands` indicates commands that were sent to the device to achieve a state. This module, as far as I can tell, is only informational in purpose (and should probably instead be part of a facts module?)

**Review by Qalthos — COMMENTED** [review]
Unless I'm mistaken, this module doesn't do anything, and should probably either turn into a facts module or a lookup plugin at a stretch.

Alternately, I suppose it could be added as an additional return key in `asa_og`, though I don't know if that module would easily support the change.

File: lib/ansible/modules/network/asa/asa_ogcontent.py (hunk 1 above)
Comment (lvrfrc87) [reply lib/ansible/modules/network/asa/asa_ogcontent.py:39]: @Qalthos you are right. That was my concern at the begin of this PR: I was not sure if there was a better way to implement this kind of module. I am ok to turn in a fact module but what would be the best way to do it? What would I need to change? Just a few indications would be great! Thanks

**Comment by FedericoOlivieri** [comment]
I ll rewrite the module to be more get_fact fashion. I close this PR and raise a new one once done

**Pull request closed without merging**